File: 2997. Minimum Number of Operations to Make Array XOR Equal to K.java
class Solution {
    public int minOperations(int[] nums, int k) {
        int answer = 0;
        for (int i = 0; i < 32; i++) {
            int count = 0;
            for (int x : nums) count = count + (check(x, i) ? 1 : 0);
            answer = answer + (((count%2 != 0) ^ check(k, i)) ? 1 : 0);
        }
        return answer;
    }
    private boolean check(int i, int pos) {
        return (i&(1<<pos)) > 0;
    }
}

File: 2997. Minimum Number of Operations to Make Array XOR Equal to K.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        answer = 0
        for i in range(0, 32):
            count = 0
            for v in nums:
                if (v & (1<<i)) > 0:
                    count += 1
            if ((count%2)>0) != ((k&(1<<i))>0):
                answer += 1
        return answer